### ISSUE TITLE
Merging manifests with different sample rate datasets

### ISSUE THREAD

**Issue by SiddGururani** [SeanNaren/deepspeech.pytorch#115]
I'm planning to run train a deepspeech model using Switchboard and Librispeech together but they're both different sample rate audio files (8k and 16k respectively).

Currently, I think I'd have to batch resample one of them (most likely Switchboard since that's the smaller one) to either 8 or 16k for them to be usable together, right? This appears to be the case since `torchaudio.load()` loads the file in the native sample rate of the file.

If I remember correctly, at some point `librosa.load()` used to be the audio file loader and that supports resampling. Using librosa might be a way to make sure that the `--sample_rate` argument is enforced upon all audio files but I believe there will be a hit on the performance.

How would you recommend using two or more datasets with different sample rate audio files?

**Comment by ryanleary** [comment]
May be easiest to just preprocess switchboard out-of-band as a separate step and then point to that.

**Comment by SiddGururani** [comment]
Umm, can you please explain what you mean by preprocess switchboard out-of-band?

Also, how much impact do you think sample rate will have during the inference step? I know that the sample rate of the data that I'll eventually run inferences on will be 8k. Should I be training the model on 8k data or is it fine to upsample the 8k test data to 16k and run the prediction? I tried doing that with librispeech test data and it was fine but as soon as I gave it some other 8k file of a phone call, it completely sucked. 
The model was trained using librispeech 1000hrs training set (800*5 GRU for 30 epochs, WER: ~21 on all librispeech validation data) and I used `3-gram.pruned.1e-7` LM from openslr.

**Comment by SeanNaren** [comment]
Baidu reports that they used 16khz, any audio that was 8k got upsampled in their paper. so should be fine to do so!

Just to re-iterate im not sure that librispeech alone is a good enough dataset to make a production model. if you really want a lot of data, it might be worth looking into an LDC licence but that is not cheap sadly.

**Comment by SiddGururani** [comment]
The only reason I feel a little hesitant to train on 16k and use it to predict using 8k audio is that it's possible for the model to learn from spectral content above 4kHz which is absent in 8k audio since it's bandlimited and this might cause issues with the inference, BUT this is only speculation.

I agree that Librispeech alone might be too clean and specific a dataset to be used in production. I'm trying to see if I can combine all the data (librispeech+switchboard+TED-LIUM+voxforge) and achieve better results. Have you guys trained using all the available datasets? If yes, I'd be curious to know how much of an improvement you observed. I believe Baidu put up a plot of WER vs hours of training data but I'd be interested to know if any of you have first-hand experience training with the datasets I mentioned.

**Comment by srviest** [comment]
@SiddGururani What results did you get using Switchboard? I always got my gradients exploded on Switchboard with multiple sets of  hyperparameters...
Could you share your experience on that? Or any suggestions would be appreciated...

**Comment by jinserk** [comment]
I'm almost close to a beginner, but I'm training the model using librispeech + tedlium + fisher (= ~3000 hrs) with 8kHz downsampling preprocess. Compared to the tedlium-only training, the WER or CER is very bad (~90%) during the training (with augment) but the decoded result with KenLM is quite impressive. My purpose is not obtaining of SOTA performance but obtaining a practical performance over the phone speaking, so it might not be a reference to you. But my previous test shows the performances from 8k and 16k are not so different.
